0029-loopback-transport/0029-loopback-transport.md

```diff
@@ -0,0 +1,33 @@
+---
+title: The Loopback Transport Protocol (LT)
+draft: 1
+---
+# Loopback Transport Protocol (LT)
+
+The Loopback Transport protocol is a technique for transporting value over the Interledger. It relies on the establishment of a temporary, direct, ILPv4 link between sender and receiver (the loopback link). The flow is as follows.
+
+## Flow
+
+The following flow describes how the sender's loopback transfer module ("LT module") interacts with the application layer protocol, the receiver, and the ILPv4 network.
+
+1. The loopback connection is established out of band. For instance, the application layer protocol may allow the receiver to specify a [BTP/2.0](../0023-bilateral-transfer-protocol/0023-bilateral-transfer-protocol.md) URL for the sender to connect to.
```

Comment:
This makes it sound like a BTP URL is just one of multiple options, is that true? Would clients / servers be expected to support other ways of connecting?

Reply:
Ok maybe I'm misunderstanding what is in scope for this protocol then. Are you saying that this requires an authenticated, encrypted, connection-oriented, framed communication protocol but that senders and receivers can choose whatever protocol they want that provides all of those features? Where does the protocol negotiation happen?

Can implementors make any assumptions about what communication protocol others will support?

> It's the same as for how sender and connector talk to each other, it's a local decision.

Actually it's very different, precisely because it's not a local decision. I have one connector that I talk to and we agree on the protocol when we peer. There are many receivers I want to pay in the world and in order to make this protocol work I need to be able to communicate to each of them. We all need to coordinate about what communication protocols we're going to support and expect other people to support.

Reply:
If those are the default options, are implementations expected to include both? If you try to connect to one that uses an incompatible protocol or version, how do you know and what error if any should you expect?

In general, if you design a protocol that has multiple options you need to specify how the negotiation happens and what the errors are if incompatible versions try talking to one another.

Reply:
Discovering the ledger-layer details of how the sender can connect to the receiver is described in the follow-up PR, #413. Let's continue this discussion there!